### ISSUE TITLE
False positive `UnusedIssueHandlerSuppression` when using `UndefinedFunction` with `referencedFunction`

### ISSUE THREAD

**Issue by susnux** [vimeo/psalm#11824]
The psalm config looks like this:

```xml
<!-- ... -->
	<issueHandlers>
		<UndefinedFunction>
			<errorLevel type="suppress">
				<referencedFunction name="p"/>
				<referencedFunction name="emit_css_tag"/>
			</errorLevel>
		</UndefinedFunction>
	</issueHandlers>
```

With psalm v5 everything worked fine, but when updating to psalm v6.16 psalm reports `UnusedIssueHandlerSuppression` for `UndefinedFunction`.
But this is wrong, see the screenshots:
- By default I get this `unusedIssueHandlerSuppression` error
- With the issue handler removed the issues are no longer suppressed and psalm reports them
- When replacing `referencedFunction` with `directory` it works - but this will suppress potentially valid cases:

| Default config | Removed issue handler | With `directory` instead |
|---|---|---|
|<img width="1847" height="1412" alt="Image" src="https://github.com/user-attachments/assets/6b5e11f1-eb0b-4b43-a915-b808b1239939" />|<img width="1869" height="1536" alt="Image" src="https://github.com/user-attachments/assets/a1e614f3-8f16-40bf-bf44-423a36300336" />|<img width="1535" height="1412" alt="Image" src="https://github.com/user-attachments/assets/80b59279-32a0-4ebf-a72f-db0e864dae23" />|
